robot_sf/gym_env/base_env.py

```
@@ -0,0 +1,113 @@
+from gymnasium import Env
+from robot_sf.gym_env.env_config import EnvSettings
+from robot_sf.render.sim_view import SimulationView, VisualizableSimState
+from robot_sf.sim.simulator import init_simulators
+
+
+class BaseEnv(Env):
+    """Base environment class that handles common functionality."""
+
+    def __init__(
+        self,
+        env_config: EnvSettings = EnvSettings(),
```

Ruff (0.8.2)

Do not perform function call `EnvSettings` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)